File: app/src/main/java/com/nuc/omeletteinputmethod/util/SymbolsManager.java
package com.nuc.omeletteinputmethod.util;

import android.content.Context;
import android.util.Log;


import java.io.IOException;
import java.util.Arrays;
import java.util.List;

/**
 * Created by admin on 2015/10/30.
 */
public class SymbolsManager {

    public String[] SMILE;
    public String[] MATH;
    public String[] BU_SHOU;
    public String[] SPECIAL;
    public String[] NET;
    public String[] RUSSIAN;
    public String[] NUMBER;
    public String[] PHONETIC;
    public String[] BOPOMOFO;
    public String[] JAPANESE;
    public String[] GREECE;

    public Symbols symbols;
    private  IniAnalysis iniAnalysis;
    
    private String[] convertValues(List<String> list) {
        return StringUtil.convertListToString(list);
    }

    public SymbolsManager(Context context){
        iniAnalysis = new IniAnalysis(context);
        symbols = new Symbols();
        try {
            SMILE       = convertValues(iniAnalysis.getValuesFromFile(symbols.SMILE));
            MATH        = convertValues(iniAnalysis.getValuesFromFile(symbols.MATH));
            BU_SHOU     = convertValues(iniAnalysis.getValuesFromFile(symbols.BU_SHOU));
            SPECIAL     = convertValues(iniAnalysis.getValuesFromFile(symbols.SPECIAL));
            NET         = convertValues(iniAnalysis.getValuesFromFile(symbols.NET));
            RUSSIAN     = convertValues(iniAnalysis.getValuesFromFile(symbols.RUSSIAN));
            PHONETIC    = convertValues(iniAnalysis.getValuesFromFile(symbols.PHONETIC));
            NUMBER      = convertValues(iniAnalysis.getValuesFromFile(symbols.NUMBER));
            BOPOMOFO    = convertValues(iniAnalysis.getValuesFromFile(symbols.BOPOMOFO));
            JAPANESE    = convertValues(iniAnalysis.getValuesFromFile(symbols.JAPNAESE));
            GREECE      = convertValues(iniAnalysis.getValuesFromFile(symbols.GREECE));
        } catch (IOException e) {
            Log.d("WIVE","lightViewAnimate"+e.toString());
            e.printStackTrace();
        }

    }

    public class Symbols{
        public String SMILE     = "symbols/smile.ini";
        public String MATH      = "symbols/shu_xue.ini";
        public String BU_SHOU   = "symbols/bu_shou.ini";
        public String SPECIAL   = "symbols/te_shu.ini";
        public String NET       = "symbols/wang_luo.ini";
        public String RUSSIAN   = "symbols/e_wen.ini";
        public String NUMBER    = "symbols/xu_hao.ini";
        public String PHONETIC  = "symbols/yin_biao.ini";
        public String BOPOMOFO  = "symbols/zhu_yin.ini";
        public String JAPNAESE  = "symbols/ri_wen_pian_jia.ini";
        public String GREECE    = "symbols/greece.ini";
    }

    @Override
    public String toString() {
        return "SymbolsManager{" +
                "SMILE=" + Arrays.toString(SMILE) +
                ", MATH=" + Arrays.toString(MATH) +
                ", BU_SHOU=" + Arrays.toString(BU_SHOU) +
                ", SPECIAL=" + Arrays.toString(SPECIAL) +
                ", NET=" + Arrays.toString(NET) +
                ", RUSSIAN=" + Arrays.toString(RUSSIAN) +
                ", NUMBER=" + Arrays.toString(NUMBER) +
                ", PHONETIC=" + Arrays.toString(PHONETIC) +
                ", BOPOMOFO=" + Arrays.toString(BOPOMOFO) +
                ", JAPANESE=" + Arrays.toString(JAPANESE) +
                ", GREECE=" + Arrays.toString(GREECE) +
                ", symbols=" + symbols +
                ", iniAnalysis=" + iniAnalysis +
                '}';
    }
}

File: app/src/main/java/com/nuc/omeletteinputmethod/util/SymbolsManager.kt
package com.nuc.omeletteinputmethod.util

import android.content.Context
import android.util.Log
import dagger.hilt.android.qualifiers.ApplicationContext
import java.io.IOException
import javax.inject.Inject

class SymbolsManager @Inject constructor(
    @ApplicationContext private val context: Context
) {

    lateinit var SMILE: Array<String>
    lateinit var MATH: Array<String>
    lateinit var BU_SHOU: Array<String>
    lateinit var SPECIAL: Array<String>
    lateinit var NET: Array<String>
    lateinit var RUSSIAN: Array<String>
    lateinit var NUMBER: Array<String>
    lateinit var PHONETIC: Array<String>
    lateinit var BOPOMOFO: Array<String>
    lateinit var JAPANESE: Array<String>
    lateinit var GREECE: Array<String>

    lateinit var symbols: Symbols
    private lateinit var iniAnalysis: IniAnalysis

    private fun convertValues(list: List<String>): Array<String> {
        return list.flatMap { it.split("\n").filter { s -> s.isNotEmpty() } }.toTypedArray()
    }

    init {
        iniAnalysis = IniAnalysis(context)
        symbols = Symbols()
        try {
            SMILE = convertValues(iniAnalysis.getValuesFromFile(symbols.SMILE))
            MATH = convertValues(iniAnalysis.getValuesFromFile(symbols.MATH))
            BU_SHOU = convertValues(iniAnalysis.getValuesFromFile(symbols.BU_SHOU))
            SPECIAL = convertValues(iniAnalysis.getValuesFromFile(symbols.SPECIAL))
            NET = convertValues(iniAnalysis.getValuesFromFile(symbols.NET))
            RUSSIAN = convertValues(iniAnalysis.getValuesFromFile(symbols.RUSSIAN))
            PHONETIC = convertValues(iniAnalysis.getValuesFromFile(symbols.PHONETIC))
            NUMBER = convertValues(iniAnalysis.getValuesFromFile(symbols.NUMBER))
            BOPOMOFO = convertValues(iniAnalysis.getValuesFromFile(symbols.BOPOMOFO))
            JAPANESE = convertValues(iniAnalysis.getValuesFromFile(symbols.JAPANESE))
            GREECE = convertValues(iniAnalysis.getValuesFromFile(symbols.GREECE))
        } catch (e: IOException) {
            Log.d("WIVE", "lightViewAnimate" + e.toString())
            e.printStackTrace()
            // Fall back to empty arrays if loading fails
            val emptyArr = emptyArray<String>()
            SMILE = emptyArr; MATH = emptyArr; BU_SHOU = emptyArr; SPECIAL = emptyArr
            NET = emptyArr; RUSSIAN = emptyArr; NUMBER = emptyArr; PHONETIC = emptyArr
            BOPOMOFO = emptyArr; JAPANESE = emptyArr; GREECE = emptyArr
        }
    }

    class Symbols {
        val SMILE = "symbols/smile.ini"
        val MATH = "symbols/shu_xue.ini"
        val BU_SHOU = "symbols/bu_shou.ini"
        val SPECIAL = "symbols/te_shu.ini"
        val NET = "symbols/wang_luo.ini"
        val RUSSIAN = "symbols/e_wen.ini"
        val NUMBER = "symbols/xu_hao.ini"
        val PHONETIC = "symbols/yin_biao.ini"
        val BOPOMOFO = "symbols/zhu_yin.ini"
        val JAPANESE = "symbols/ri_wen_pian_jia.ini"
        val GREECE = "symbols/greece.ini"
    }

    override fun toString(): String {
        return "SymbolsManager{" +
                "SMILE=" + SMILE.contentToString() +
                ", MATH=" + MATH.contentToString() +
                ", BU_SHOU=" + BU_SHOU.contentToString() +
                ", SPECIAL=" + SPECIAL.contentToString() +
                ", NET=" + NET.contentToString() +
                ", RUSSIAN=" + RUSSIAN.contentToString() +
                ", NUMBER=" + NUMBER.contentToString() +
                ", PHONETIC=" + PHONETIC.contentToString() +
                ", BOPOMOFO=" + BOPOMOFO.contentToString() +
                ", JAPANESE=" + JAPANESE.contentToString() +
                ", GREECE=" + GREECE.contentToString() +
                ", symbols=" + symbols +
                ", iniAnalysis=" + iniAnalysis +
                '}'
    }
}